tap_appstore/streams.py

```diff
@@ -1,72 +1,150 @@
 """Stream type classes for tap-appstore."""
 
 from __future__ import annotations
+import tempfile
 
-import sys
-import typing as t
+import os
+from applaud.connection import Connection
+from applaud.endpoints.sales_reports import SalesReportsEndpoint
+from singer_sdk import Stream, typing as th
+import datetime
+from datetime import datetime
+import logging
 
-from singer_sdk import typing as th  # JSON Schema typing helpers
+logger = logging.getLogger(__name__)
 
-from tap_appstore.client import AppStoreStream
 
-if sys.version_info >= (3, 9):
-    import importlib.resources as importlib_resources
-else:
-    import importlib_resources
+class SalesReportStream(Stream):
+    name = "sales_reports"
+    #primary_keys = ["id"]
+    schema = th.PropertiesList(
+        th.Property("Provider", th.StringType),
+        th.Property("Provider Country", th.StringType),
+        th.Property("SKU", th.StringType),
+        th.Property("Developer", th.StringType),
+        th.Property("Title", th.StringType),
+        th.Property("Version", th.StringType),
+        th.Property("Product Type Identifier", th.StringType),
+        th.Property("Units", th.IntegerType),
+        th.Property("Developer Proceeds", th.NumberType),
+        th.Property("Begin Date", th.DateTimeType),
+        th.Property("End Date", th.DateTimeType),
+        th.Property("Customer Currency", th.StringType),
+        th.Property("Country Code", th.StringType),
+        th.Property("Currency of Proceeds", th.StringType),
+        th.Property("Apple Identifier", th.StringType),
+        th.Property("Customer Price", th.NumberType),
+        th.Property("Promo Code", th.StringType),
+        th.Property("Parent Identifier", th.StringType),
+        th.Property("Subscription", th.StringType),
+        th.Property("Period", th.StringType),
+        th.Property("Category", th.StringType),
+        th.Property("CMB", th.StringType),
+        th.Property("Device", th.StringType),
+        th.Property("Supported Platforms", th.StringType),
+        th.Property("Proceeds Reason", th.StringType),
+        th.Property("Preserved Pricing", th.StringType),
+        th.Property("Client", th.StringType),
+        th.Property("Order Type", th.StringType)
+    ).to_dict()
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.get_records()
```

Comment:
We don't need to call the `get_records` here, it's already called in sync records:
https://github.com/meltano/sdk/blob/d5b051508185f895de74b6fb84883b18504cb61d/singer_sdk/streams/core.py#L1033

Reply:
Ah, TIL! Thank you. Fixed. ✅